### PR TITLE
Deploy n readme

### PR DESCRIPTION
**Pull request by ffilippopoulos**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,125 +1,148 @@
...
-$ make local-envoy
-$ make local-envoy-mac # on macOS
-```
+### Sources configuration - control plane
 
-You can inspect the current Envoy config by visting `/config_dump`
-```
-$ curl http://localhost:9901/config_dump
-```
+Each source is a kubernets cluster that the control plane is going to watch events from. An example config will be:
```
Comment (ribbybibby) [inline README.md]: kubernetes*

File: README.md [hunk 2]
```diff
@@ -1,125 +1,148 @@
...
-2019/06/27 09:20:45 [DEBUG] Response: responding (1) with type: type.googleapis.com/envoy.api.v2.Listener, version: 2019-06-27 01:20:45.705185 +0100 BST m=+240.565257900, resources: 1
 ```
 
-Check `http://localhost:9901/config_dump` to see the listeners and clusters that have been added to Envoy by the control plane.
+You need to supply a list of source and the location of the respective kubeconfig files.
+`listenersource` defines the primary kubernetes cluster, meaning the cluster which the control plane will watch for listeners. It should be set to true only for 1 cluster, the same one that the control plane is deployed. That way envoy sidecars in each kubernetes cluster will talk to the local control plane to fetch config, listeners will only be defined in the same cluster where the envoy sidecar is running and clusters will be generated from all the available sources.
+Losing networking between clusters means that remote cluster members will disappear but everything will keep working for the local cluster.
```
Comment (ribbybibby) [inline README.md]: If we don't already, could we create a check in the code to make sure there is only one 'primary' kubernetes cluster?

File: README.md [hunk 3]
```diff
@@ -1,125 +1,148 @@
...
-  nodename: test-app
-  listenport: 8080
-  targetport: 8081
-  rbacallowcluster: test-cluster
+  nodename: <envoy-node-name>
+  listenport: 8080 # Port to listen for ingress traffic
+  targetport: 80 # Port to forward on localhost
+  rbac:
+    cluster: <cluster> # Only accept incoming traffic from cluster
+    Sans: # List of SANs to accept traffic from in case of tls
```
Comment (ribbybibby) [inline README.md]: `Sans:` is capitalised but the other options aren't.

File: README.md [hunk 4]
```diff
@@ -1,125 +1,148 @@
...
-  listenport: 9090
-  targetport: 8080
-  targetcluster: test-cluster
-  lbpolicy: tcp
+  nodename: <envoy-node-name>
+  listenport: 8080 # Port to listen on localhost for egress traffic.
+  target:
+    cluster: <cluster> # The target cluster to forward traffic to
+    port: 8080 # Target cluster port to send traffic
+  lbpolicy: "tcp" # loab balancing policy (tcp|http), default is tcp
```
Comment (ribbybibby) [inline README.md]: `load`*

File: README.md [hunk 5]
```diff
@@ -1,125 +1,148 @@
...
 spec:
-  nodename: test-app
-  listenport: 8080
-  targetport: 8081
-  rbacallowcluster: test-cluster
+  nodename: <envoy-node-name>
+  listenport: 8080 # Port to listen for ingress traffic
+  targetport: 80 # Port to forward on localhost
+  rbac:
+    cluster: <cluster> # Only accept incoming traffic from cluster
```
Comment (ribbybibby) [inline README.md]: Not really a review of this PR but `cluster` probably needs to be a list, to support multiple upstream services.

File: deploy/example/control-plane/kustomization.yaml [hunk 6]
```diff
@@ -0,0 +1,17 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+bases:
+  - ../../base/
+  - ../../cluster/
+resources:
+  # need to define a configMap with the confi for the control plane
```
Comment (ribbybibby) [inline deploy/example/control-plane/kustomization.yaml]: `config`*

File: deploy/example/control-plane/kustomization.yaml [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+  - config.yaml
+  # Secrets for the kubeconfig of each cluster are neeeded
+  # e.t.c.
```
Comment (ribbybibby) [inline deploy/example/control-plane/kustomization.yaml]: Should be `e.g`.

**Review by ribbybibby — APPROVED** [review]
Looks good to me. Comments are all spelling/grammar nazism and some thinking out loud from me.

File: README.md (hunk 3 above)
Comment (ffilippopoulos) [reply README.md]: that is the most interesting one, as far as I can see `sans` doesn't really work so this should be checked in code first

File: README.md (hunk 2 above)
Comment (ffilippopoulos) [reply README.md]: Rethinking this one, I believe we should allow as many `listenersource` cluster as the user wants. It makes sense for how we deploy to have only one, but I can imagine setups where there you run the control plane in one cluster and others fetch listener configuration from there. Let's leave it until it becomes an issue??

Comment (ribbybibby) [reply README.md]: :+1:

**Pull request merged**